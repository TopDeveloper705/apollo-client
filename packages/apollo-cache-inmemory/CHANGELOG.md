# Change log

### vNEXT
- Refactored internal store from apollo-client into own package
