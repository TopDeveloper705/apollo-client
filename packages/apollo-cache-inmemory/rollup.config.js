import build from '../../config/rollup.config';

export default build('apollo.cache.inmemory');
