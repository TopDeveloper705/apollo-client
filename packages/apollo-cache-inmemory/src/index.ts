export {
  InMemoryCache,
  InMemoryCacheConfig,
  defaultDataIdFromObject,
} from './inMemoryCache';

export * from './readFromStore';
export * from './writeToStore';
export * from './objectCache';
export * from './types';
export * from './references';
