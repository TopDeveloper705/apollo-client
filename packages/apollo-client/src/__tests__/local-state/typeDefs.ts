import { InMemoryCache } from 'apollo-cache-inmemory';
import { ApolloLink, Observable } from 'apollo-link';
import gql from 'graphql-tag';

import ApolloClient from '../..';

describe('Default use', () => {
  it(
    'should add a schema string in SDL format to the context as definition ' +
      'if typeDefs are passed in',
    done => {
      const link = new ApolloLink(operation => {
        const { schemas } = operation.getContext();
        expect(schemas).toMatchSnapshot();
        return Observable.of({
          data: { foo: { bar: true, __typename: 'Bar' } },
        });
      });

      const typeDefs = `
        type Todo {
          id: String
          message: String!
        }

        type Query {
          todo(id: String!): Todo
        }
      `;

      const remoteQuery = gql`
        {
          foo {
            bar
          }
        }
      `;

      const client = new ApolloClient({
        cache: new InMemoryCache(),
        link,
        typeDefs,
      });

      client.query({ query: remoteQuery }).then(() => done(), done.fail);
    },
  );

  it('should concatenate schema strings if typeDefs are passed in as an array', done => {
    const anotherSchema = `
        type Foo {
          foo: String!
          bar: String
        }
      `;

    const link = new ApolloLink(operation => {
      const { schemas } = operation.getContext();
      expect(schemas).toMatchSnapshot();
      return Observable.of({
        data: { foo: { bar: true, __typename: 'Bar' } },
      });
    });

    const typeDefs = `
        type Todo {
          id: String
          message: String!
        }

        type Query {
          todo(id: String!): Todo
        }
      `;

    const remoteQuery = gql`
      {
        foo {
          bar
        }
      }
    `;

    const client = new ApolloClient({
      cache: new InMemoryCache(),
      link,
      typeDefs: [typeDefs, anotherSchema],
    });

    client.query({ query: remoteQuery }).then(() => done(), done.fail);
  });
});
