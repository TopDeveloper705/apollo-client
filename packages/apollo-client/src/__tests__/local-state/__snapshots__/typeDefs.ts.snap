// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default use should add a schema string in SDL format to the context as definition if typeDefs are passed in 1`] = `
Array [
  Object {
    "definition": "type Todo {
          id: String
          message: String!
        }

        type Query {
          todo(id: String!): Todo
        }",
    "directives": "directive @client on FIELD",
  },
]
`;

exports[`Default use should concatenate schema strings if typeDefs are passed in as an array 1`] = `
Array [
  Object {
    "definition": "type Todo {
          id: String
          message: String!
        }

        type Query {
          todo(id: String!): Todo
        }
type Foo {
          foo: String!
          bar: String
        }",
    "directives": "directive @client on FIELD",
  },
]
`;
